pkg/upgrade/drain.go

```diff
@@ -20,12 +20,14 @@ import (
 )
 
 func (u *simpleUpgrader) drain(ctx context.Context, cs *api.OpenShiftManagedCluster, role api.AgentPoolProfileRole, nodeName string) error {
-	kc, err := managedcluster.ClientsetFromV1Config(cs.Config.AdminKubeconfig)
-	if err != nil {
-		return err
+	var err error
+	if u.kubeclient == nil {
```

Comment:
I would prefer to see us set this during the construction of simpleUpgrader as opposed to doing it here. It keeps the logic of `drain` clearer and does not litter the code with conditionals that are helping unit testing which is not ideal.

Reply:
Setting up the client does not require any sort of connection, you just need a kubeconfig and you have that.

Reply:
from slack: "I think for now it's probably fine to do what you started doing

and we can probably clean it up as part of https://github.com/openshift/openshift-azure/issues/602